src/clj/rems/api/user_settings.clj

```diff
@@ -0,0 +1,31 @@
+(ns rems.api.user-settings
+  (:require [compojure.api.sweet :refer :all]
+            [rems.api.schema :refer :all]
+            [rems.db.user-settings :as user-settings]
+            [rems.util :refer [getx-user-id]]
+            [ring.util.http-response :refer :all]
+            [schema.core :as s]))
+
+(s/defschema UpdateUserSettingsCommand
+  {:user-id s/Str
+   (s/optional-key :language) s/Keyword})
+
+(s/defschema UserSettings
+  {(s/optional-key :language) s/Keyword})
+
+(def user-settings-api
+  (context "/user-settings" []
+    :tags ["user-settings"]
+
+    (GET "/" []
+      :summary "Get user settings"
+      :roles #{:logged-in}
+      :return UserSettings
+      (ok (user-settings/get-user-settings (getx-user-id))))
+
+    (PUT "/update" []
+      :summary "Update user settings"
+      :roles #{:logged-in}
+      :body [command UpdateUserSettingsCommand]
+      :return SuccessResponse
+      (ok (user-settings/update-user-settings! (:user-id command) command)))))
```

Comment:
This allows anybody updating any other user's settings. The user-id parameter should probably be removed from the command, so that it'll only change the current user's settings.

Reply:
Good point, done.